angular-chess/src/app/services/move-execution.service.ts

```diff
@@ -74,6 +74,10 @@ export class MoveExecutionService {
   }
 
   private finishMove(move: Move): void {
+    if (!(move.piece.type === PieceType.PAWN && Math.abs(move.from.row - move.to.row) === 2)) {
+      this.boardService.clearEnPassantSquares();
+    }
+
     if (!(move.isShortCastle || move.isLongCastle)) {
```

Comment:
Optional: refactoring hint: you can extract this condition to extra method of Move: IsCastle